src/box/lua/schema.lua

```diff
@@ -1078,8 +1078,134 @@ end
+    space_mt.update = function(space, key, ops)
+        check_space_arg(space, 'update')
+        return check_primary_index(space):update(key, ops)
+    end
+    space_mt.upsert = function(space, tuple_key, ops, deprecated)
+        check_space_arg(space, 'upsert')
+        if deprecated ~= nil then
+            local msg = "Error: extra argument in upsert call: "
+            msg = msg .. tostring(deprecated)
+            msg = msg .. ". Usage :upsert(tuple, operations)"
+            box.error(box.error.PROC_LUA, msg)
+        end
+        return internal.upsert(space.id, tuple_key, ops);
+    end
+    space_mt.delete = function(space, key)
+        check_space_arg(space, 'delete')
+        return check_primary_index(space):delete(key)
+    end
+    -- Assumes that spaceno has a TREE (NUM) primary key
+    -- inserts a tuple after getting the next value of the
+    -- primary key and returns it back to the user
+    space_mt.auto_increment = function(space, tuple)
+        check_space_arg(space, 'auto_increment')
+        local max_tuple = check_primary_index(space):max()
+        local max = 0
+        if max_tuple ~= nil then
+            max = max_tuple[1]
+        end
+        table.insert(tuple, 1, max + 1)
+        return space:insert(tuple)
+    end
+
+    space_mt.pairs = function(space, key, opts)
+        check_space_arg(space, 'pairs')
+        local pk = space.index[0]
+        if pk == nil then
+            -- empty space without indexes, return empty iterator
+            return fun.iter({})
+        end
+        return pk:pairs(key, opts)
+    end
+    space_mt.__pairs = space_mt.pairs -- Lua 5.2 compatibility
+    space_mt.__ipairs = space_mt.pairs -- Lua 5.2 compatibility
+    space_mt.truncate = function(space)
+        check_space_arg(space, 'truncate')
+        return internal.truncate(space.id)
+    end
+    space_mt.format = function(space, format)
+        check_space_arg(space, 'format')
+        return box.schema.space.format(space.id, format)
+    end
+    space_mt.drop = function(space)
+        check_space_arg(space, 'drop')
+        check_space_exists(space)
+        return box.schema.space.drop(space.id, space.name)
+    end
+    space_mt.rename = function(space, name)
+        check_space_arg(space, 'rename')
+        check_space_exists(space)
+        return box.schema.space.rename(space.id, name)
+    end
+    space_mt.create_index = function(space, name, options)
+        check_space_arg(space, 'create_index')
+        check_space_exists(space)
+        return box.schema.index.create(space.id, name, options)
+    end
+    space_mt.run_triggers = function(space, yesno)
+        check_space_arg(space, 'run_triggers')
+        local s = builtin.space_by_id(space.id)
+        if s == nil then
+            box.error(box.error.NO_SUCH_SPACE, space.name)
+        end
+        builtin.space_run_triggers(s, yesno)
+    end
+
+    return space_mt
+end)()
+
+box.index_mt = (function()
```

Comment:
We do not use in tarantool functions that are called right after creation. Please, initialise box.index_mt with no closures. For example, like this:
```Lua
local index_mt = {}
local function index_mt:len()
-- 'index' here can be accessed as 'self'.
end
local function index_mt:min_ffi(key)
-- ..
end
--..
--..
box.schema.index_mt = { __index = index_mt }
```